kotlinx-coroutines-core/common/test/ImmediateYieldTest.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+import kotlin.coroutines.*
+import kotlin.test.*
+
+class ImmediateYieldTest : TestBase() {
+
+    // See https://github.com/Kotlin/kotlinx.coroutines/issues/1474
+    @Test
+    fun testImmediateYield() = runTest {
```

Comment:
It would be nice to have a more realistic test as well. 
E.g. like `HandlerDispatcherTest` and `JavaFxTest`

Reply:
Added tests for Android, JavaFx, and Swing